migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,310 @@
+	batchSize = 20
+)
+
+func createV2reportConfig(reportConfigProto *storage.ReportConfiguration) *storage.ReportConfiguration {
+	// clone v1 report config
+	newConfig := reportConfigProto.Clone()
+	// populate id
+	id := getDeterministicID(reportConfigProto.GetId())
+	newConfig.Id = id
+	// assign collection id in resource scope
+	newConfig.ResourceScope = &storage.ResourceScope{
+		ScopeReference: &storage.ResourceScope_CollectionId{CollectionId: reportConfigProto.GetScopeId()},
+	}
+	// set scope id to empty string so that v2 api does filter out v2 configs
+	newConfig.ScopeId = ""
+	// add vuln report filter to v2 copy of report config
+	vulnFilter := &storage.VulnerabilityReportFilters{
+		Severities: reportConfigProto.GetVulnReportFilters().GetSeverities(),
+		Fixability: reportConfigProto.GetVulnReportFilters().GetFixability(),
+		ImageTypes: []storage.VulnerabilityReportFilters_ImageType{storage.VulnerabilityReportFilters_DEPLOYED},
+	}
+	if reportConfigProto.GetVulnReportFilters().GetSinceLastReport() {
+		vulnFilter.CvesSince = &storage.VulnerabilityReportFilters_SinceLastSentScheduledReport{
+			SinceLastSentScheduledReport: true,
+		}
+	} else {
+		vulnFilter.CvesSince = &storage.VulnerabilityReportFilters_AllVuln{
+			AllVuln: true,
+		}
+	}
+	newConfig.Filter = &storage.ReportConfiguration_VulnReportFilters{
+		VulnReportFilters: vulnFilter,
+	}
+	return newConfig
+}
+
+func createNotifier(reportConfigProto *storage.ReportConfiguration) *storage.NotifierConfiguration {
+	notifierConfig := &storage.NotifierConfiguration{
+		Ref: &storage.NotifierConfiguration_Id{
+			Id: reportConfigProto.GetEmailConfig().GetNotifierId()},
+		NotifierConfig: &storage.NotifierConfiguration_EmailConfig{
+			EmailConfig: &storage.EmailNotifierConfiguration{
+				MailingLists: reportConfigProto.GetEmailConfig().GetMailingLists(),
+			},
+		}}
+	return notifierConfig
+}
+
+func createReportSnapshot(v1Config *storage.ReportConfiguration, v2Config *storage.ReportConfiguration) *storage.ReportSnapshot {
+
+	// create report snapshot for last successful scheduled report job
+	if v1Config.GetLastSuccessfulRunTime() == nil {
+		return nil
+	}
+	if v1Config.GetLastRunStatus() != nil {
+		return &storage.ReportSnapshot{
+			ReportConfigurationId: v2Config.GetId(),
+			Name:                  v2Config.GetName(),
+			Description:           v2Config.GetDescription(),
+			Type:                  storage.ReportSnapshot_VULNERABILITY,
+			ReportId:              uuid.NewV4().String(),
+			Collection: &storage.CollectionSnapshot{
+				Id: v2Config.GetResourceScope().GetCollectionId(),
+			},
+			Schedule: v2Config.GetSchedule(),
+			ReportStatus: &storage.ReportStatus{
+				RunState:          storage.ReportStatus_DELIVERED,
+				ReportRequestType: storage.ReportStatus_SCHEDULED,
+				CompletedAt:       v1Config.LastSuccessfulRunTime,
+			},
+		}
+	}
+	return nil
+}
+
+func checkifNotifierExsists(notifierID string, db *gorm.DB, dbctx context.Context) (bool, error) {
+
```

Comment:
I think you should change this and the `getMigratedReportConfigIfExsists` to be more like this:

```
// Use GORM API build SQLrow := db.Table("users").Where("name = ?", "jinzhu").Select("name", "age").Row()row.Scan(&name, &age)
```

This way you don't need a loop to get a simple thing and you don't have to worry about the call to close or the call to next.